### PR TITLE
Feature/implement card component

### PR DESCRIPTION
**Pull request by lhoursquentin**
Very basic white background container with a thin border
card-preview
We'll probably add rows/title/... subcomponents to this in the future

### CODE REVIEW

File: src/components/card/Card.module.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
+@use '../../colors/colors';
+@use '../../typography/typography';
+
+.card {
+  background: #ffffff;
```
[the remark this answers is not in the document]
Comment (lhoursquentin) [reply src/components/card/Card.module.scss:5]: ah you are right thanks for spotting that, will fix it

File: src/components/card/Card.module.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  border-radius: 8px;
+  border-style: solid;
+  border-width: 1px;
+  box-sizing: border-box;
+  color: colors.$black;
+  display: flex;
+  flex-direction: column;
+  flex-grow: 1;
+  margin: 16px;
+  min-width: 150px;
```
Comment (akamkaro) [inline src/components/card/Card.module.scss:16]: Ordered alphabetically <3

File: src/components/card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { getClassNames } from '@utility/cssUtils';
+import styles from './Card.module.scss';
+
+type CardSize = 'small' | 'large';
```
Comment (Alexandre-Herve) [inline src/components/card/Card.tsx:5]: Mmh I'm not a fan of this API at all. There will most likely be as many card sizes as designs.
I would rather have let the dev using the card decide the width and height with props, I don't really see why we would have a limited amount of sizes.

File: src/components/card/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export interface CardProps extends React.HTMLAttributes<HTMLDivElement> {
+  size?: CardSize;
+}
+
+export const Card = (props: CardProps) => {
+  return (
+    <div
+      {...props}
```
Comment (Alexandre-Herve) [inline src/components/card/Card.tsx:14]: In such a situation I would rather do something like this:
```
const { size, ...rest } = props
return <div {...rest} ...
```
Becauze `size` doesn't belong to the props for div.

Comment (Alexandre-Herve) [reply src/components/card/Card.tsx:14]: In this case though what I would rather do is this:
```
const Card = (props: React.HTMLAttributes<HTMLDivElement>) => {
 const { className, children, ...rest } = props
 const computedClassName = [className || '', styles.card].join(' ')
 return (
   <div {...rest} className={computedClassName}>{children}</div>
 )
```
Or something in that taste

File: src/components/card/Card.tsx (hunk 3 above)
Comment (lhoursquentin) [reply src/components/card/Card.tsx]: Removed sizes, let's go very basic for now

File: src/components/card/Card.module.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+@use '../../typography/typography';
+
+.card {
+  background: colors.$white;
+  border-color: colors.$dash-green-05;
+  border-radius: 8px;
+  border-style: solid;
+  border-width: 1px;
+  box-sizing: border-box;
+  color: colors.$black;
```
Comment (Alexandre-Herve) [inline src/components/card/Card.module.scss:11]: I'd have left this for the user to decide, IMO the card is more a layout component

File: src/components/card/Card.module.scss [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  display: flex;
+  flex-direction: column;
+  flex-grow: 1;
+  margin: 16px;
+  min-height: 132px;
```
Comment (Alexandre-Herve) [inline src/components/card/Card.module.scss:16]: Seems a bit arbitrary. Why enforce this?

File: src/components/card/Card.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+import { getClassNames } from '@utility/cssUtils';
+import styles from './Card.module.scss';
+
+export const Card = (props: React.HTMLAttributes<HTMLDivElement>) => {
+  const { className, children, ...rest } = props;
+  const computedClassName = getClassNames(className ?? '', styles.card);
```
Comment (Alexandre-Herve) [inline src/components/card/Card.tsx:7]: You don't need the `?? ''` anymore, I allowed `undefined` arguments in `getClassNames` lately

File: src/components/card/Card.module.scss [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  min-width: 150px;
+  padding: 16px;
+  @include typography.body-medium;
+
+  /* Content */
+
+  .content {
```
Comment (Alexandre-Herve) [inline src/components/card/Card.module.scss:23]: I don't see this `.content` used anywhere?

Comment (lhoursquentin) [reply src/components/card/Card.module.scss]: Kept it from the step-by-step guide, removed it.

File: src/components/card/Card.module.scss (hunk 5 above)
Comment (lhoursquentin) [reply src/components/card/Card.module.scss:11]: Talked offline, we might want to support dark mode later but let's keep this as is for now and use more specific styles to override it if needed.

**Pull request merged**